### PR TITLE
add blt options for gtest and fruit

### PR DESCRIPTION
**Pull request by cyrush**
Adds:

 `ENABLE_GTEST` (default = ON)
 `ENABLE_FRUIT` (default = ON (also guarded by `ENABLE_FORTRAN` = ON)

Like `ENABLE_GMOCK`, these are guarded by `ENABLE_TESTS`.

gmock is also guarded by `ENABLE_GTEST`, 
I think this makes sense b/c building gmock always builds gtest.

### CODE REVIEW

**Comment by cyrush** [comment]
This it should help with #61.

This is a WIP (need to fix tests in example),

**Comment by cyrush** [comment]
resolve testing issues, let me know if you have any suggestions

File: thirdparty_builtin/CMakeLists.txt [hunk 1]
```diff
@@ -70,27 +70,36 @@ if(ENABLE_TESTS)
...
+    
+    if(ENABLE_GTEST)
+        # Enable builtin google test 
+        add_subdirectory(googletest-release-1.8.0)
+
+        blt_register_library(NAME gtest
+                             INCLUDES ${gtest_SOURCE_DIR}/include
+                             LIBRARIES gtest_main gtest
                              DEFINES  ${gtest_defines})
+         if(ENABLE_GMOCK)
```
Comment (davidbeckingsale) [inline thirdparty_builtin/CMakeLists.txt]: Should `ENABLE_GMOCK` just force `ENABLE_GTEST`?

Comment (cyrush) [inline thirdparty_builtin/CMakeLists.txt]: Yeah, that sounds good -- either that or we should add an error message.

Comment (cyrush) [inline thirdparty_builtin/CMakeLists.txt]: @davidbeckingsale  I added logic to force ENABLE_GTEST to ON, if ENABLE_GMOCK is ON

**Comment by cyrush** [comment]
@white238, what do you think?

File: cmake/BLTOptions.cmake [hunk 2]
```diff
@@ -61,9 +61,17 @@ option(ENABLE_MPI          "Enables MPI support" OFF)
...
+# Options that control if Google Test, Google Mock, and Fruit are built 
+# and available for use. 
+#
+# If ENABLE_TESTS=OFF, no testing support is built and these option are ignored.
+#
+# Google Mock requires and always builds Google Test, so ENABLE_GMOCK=ON implies ENABLE_GTEST=ON.
+#
+option(ENABLE_GTEST        "Enable Google Test testing support (if ENABLE_TESTS=ON)" ON)
 option(ENABLE_GMOCK        "Enable Google Mock testing support (if ENABLE_TESTS=ON)" OFF)
+option(ENABLE_FRUIT        "Enable Fruit testing support (if ENABLE_TESTS=ON and ENABLE_FORTRAN=ON)" ON)
```
Comment (white238) [inline cmake/BLTOptions.cmake]: Do you check if they enable fruit but don't enable fortran?  Same with enable gtest/gmock/fruit without enabling test... I would rather error and exit then ignore an option that they probably got wrong.

Comment (cyrush) [reply cmake/BLTOptions.cmake]: reviewed with @white238, the fact that these are guarded in a nested manner ( fruit support requires ENABLE_FORTRAN and ENABLE_FRUIT) means we shouldn't hit any strange cases.

**Pull request merged**